src/CsvWriter.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+namespace Keboola\Csv;
+
+class CsvWriter extends AbstractCsvFile
+{
+    /**
+     * @var string
+     */
+    private $lineBreak;
+
+    /**
+     * CsvFile constructor.
+     * @param string|resource $file
+     * @param string $delimiter
+     * @param string $enclosure
+     * @param string $lineBreak
+     * @throws Exception
+     * @throws InvalidArgumentException
+     */
+    public function __construct(
+        $file,
+        $delimiter = self::DEFAULT_DELIMITER,
+        $enclosure = self::DEFAULT_ENCLOSURE,
+        $lineBreak = "\n"
+    ) {
+        $this->setDelimiter($delimiter);
+        $this->setEnclosure($enclosure);
+        $this->setLineBreak($lineBreak);
+        $this->setFile($file);
+    }
+
+    /**
+     * @param string $lineBreak
+     */
+    private function setLineBreak($lineBreak)
+    {
+        $this->validateLineBreak($lineBreak);
+        $this->lineBreak = $lineBreak;
+    }
+
+    /**
+     * @param string $lineBreak
+     */
+    private function validateLineBreak($lineBreak)
+    {
+        $allowedLineBreaks = [
+            "\r\n", // win
+            "\r", // mac
+            "\n", // unix
+        ];
+        if (!in_array($lineBreak, $allowedLineBreaks)) {
+            throw new Exception(
+                "Invalid line break: " . json_encode($lineBreak) .
+                " allowed line breaks: " . json_encode($allowedLineBreaks),
+                Exception::INVALID_PARAM
+            );
+        }
+    }
+
+    /**
+     * @param string $fileName
+     * @throws Exception
+     */
+    protected function openCsvFile($fileName)
+    {
+        $this->filePointer = @fopen($fileName, 'w');
+        if (!$this->filePointer) {
+            throw new Exception(
+                "Cannot open file {$fileName} " . error_get_last()['message'],
+                Exception::FILE_NOT_EXISTS
+            );
+        }
+    }
+
+    /**
+     * @param array $row
+     * @throws Exception
+     */
+    public function writeRow(array $row)
+    {
+        $str = $this->rowToStr($row);
+        $ret = fwrite($this->getFilePointer(), $str);
+
+        /* According to http://php.net/fwrite the fwrite() function
+         should return false on error. However not writing the full
+         string (which may occur e.g. when disk is full) is not considered
+         as an error. Therefore both conditions are necessary. */
+        if (($ret === false) || (($ret === 0) && (strlen($str) > 0))) {
+            throw new Exception(
+                "Cannot write to CSV file " . $this->fileName .
+                ' Error: ' . error_get_last()['message'] . ' Return: ' . json_encode($ret) .
```

Comment:
Myslím že se může stát že v `error_get_last` nic nebude a pak to failne při přístupu k `null`.

Reply:
aha, no to mi uslo, ze error_get_last vraci null, protoze jsem to zkousel bez erroru. mazec:
```
php > var_export(null['test']);
php > NULL
```
takze error to neudela, ale vrchol cistoty to teda neni :) tak tam asi dam `json_encode`

Reply:
```
Accessing variables of other types (not including arrays or objects implementing the appropriate interfaces) using [] or {} silently returns NULL.
```
http://php.net/manual/en/language.types.string.php
Jsem se tam dostal přes tohle  https://stackoverflow.com/questions/10990321/why-does-php-not-complain-when-i-treat-a-null-value-as-an-array-like-this

Tohle je taky super zápis `$z = null['neco']['sdfdsf'];` :)